### ISSUE TITLE
"Caught unexcepted exception" when reading from GeoJSON

### ISSUE THREAD

**Issue by hawkaa** [PDAL/PDAL#3089]
**Describe the bug**
Hi all!

I'm trying to use PDAL to convert a GeoJSON to a LAZ and/or TIF file. However, I can't seem to get PDAL to work when reading GeoJSON, even not if the GeoJSON is produced by PDAL itself.

How to reproduce:

```
$ pdal -v 8 translate myfile.laz myfile.geojson
(PDAL Debug) Debugging...
(pdal translate readers.las Debug) GDAL debug: OGRSpatialReference::Validate: No root pointer.
(pdal translate readers.las Debug) GDAL debug: OGRSpatialReference::Validate: No root pointer.
(pdal translate readers.las Debug) GDAL debug: OGRSpatialReference::Validate: No root pointer.
(pdal translate writers.ogr Debug) GDAL debug: GNM: GNMRegisterAllInternal
(pdal translate writers.ogr Debug) GDAL debug: GNM: RegisterGNMFile
(pdal translate writers.ogr Debug) GDAL debug: GNM: RegisterGNMdatabase
(pdal translate writers.ogr Debug) GDAL debug: GNM: GNMRegisterAllInternal
(pdal translate writers.ogr Debug) GDAL debug: GNM: RegisterGNMFile
(pdal translate writers.ogr Debug) GDAL debug: GNM: RegisterGNMdatabase
(pdal translate Debug) Executing pipeline in stream mode.
(pdal translate writers.ogr Debug) GDAL debug: GDAL: GDALDriver::Create(GeoJSON,myfile.geojson,0,0,0,Unknown,0x0)
(pdal translate writers.ogr Debug) GDAL debug: GDAL: GDALClose(myfile.geojson, this=0x7f9a94d082f0)
```

Then try to convert it back to a laz file:
```
$ pdal -v 8 translate -r readers.gdal myfile.geojson myfile_copy.laz
(PDAL Debug) Debugging...
(pdal translate readers.gdal Debug) GDAL debug: OGRSpatialReference::Validate: No root pointer.
(pdal translate readers.gdal Debug) GDAL debug: OGRSpatialReference::Validate: No root pointer.
(pdal translate readers.gdal Debug) GDAL debug: OGRSpatialReference::Validate: No root pointer.
(pdal translate readers.gdal Debug) GDAL debug: GNM: GNMRegisterAllInternal
(pdal translate readers.gdal Debug) GDAL debug: GNM: RegisterGNMFile
(pdal translate readers.gdal Debug) GDAL debug: GNM: RegisterGNMdatabase
(pdal translate readers.gdal Debug) GDAL debug: GNM: GNMRegisterAllInternal
(pdal translate readers.gdal Debug) GDAL debug: GNM: RegisterGNMFile
(pdal translate readers.gdal Debug) GDAL debug: GNM: RegisterGNMdatabase
(pdal translate readers.gdal Debug) GDAL debug: GeoJSON: First pass: 1.69 %
... 
(pdal translate readers.gdal Debug) GDAL debug: GeoJSON: First pass: 99.93 %
(pdal translate readers.gdal Debug) GDAL debug: GeoJSON: First pass: 100.00 %
(pdal translate readers.gdal Debug) GDAL debug: GDAL: GDALOpen(myfile.geojson, this=0x7fa71cd2d850) succeeds as GeoJSON.
PDAL: Caught unexpected exception.

(pdal translate readers.gdal Debug) GDAL debug: GDAL: GDALClose(myfile.geojson, this=0x7fa71cd2d850)
```

Same results for `.tif`:

```
$ pdal -v 8 translate -r readers.gdal myfile.geojson myfile_copy.tif
(PDAL Debug) Debugging...
(pdal translate readers.gdal Debug) GDAL debug: OGRSpatialReference::Validate: No root pointer.
(pdal translate readers.gdal Debug) GDAL debug: OGRSpatialReference::Validate: No root pointer.
(pdal translate readers.gdal Debug) GDAL debug: OGRSpatialReference::Validate: No root pointer.
(pdal translate readers.gdal Debug) GDAL debug: GNM: GNMRegisterAllInternal
(pdal translate readers.gdal Debug) GDAL debug: GNM: RegisterGNMFile
(pdal translate readers.gdal Debug) GDAL debug: GNM: RegisterGNMdatabase
(pdal translate readers.gdal Debug) GDAL debug: GNM: GNMRegisterAllInternal
(pdal translate readers.gdal Debug) GDAL debug: GNM: RegisterGNMFile
(pdal translate readers.gdal Debug) GDAL debug: GNM: RegisterGNMdatabase
(pdal translate readers.gdal Debug) GDAL debug: GeoJSON: First pass: 1.69 %
...
(pdal translate readers.gdal Debug) GDAL debug: GeoJSON: First pass: 99.93 %
(pdal translate readers.gdal Debug) GDAL debug: GeoJSON: First pass: 100.00 %
(pdal translate readers.gdal Debug) GDAL debug: GDAL: GDALOpen(myfile.geojson, this=0x7fc5ca10ebf0) succeeds as GeoJSON.
PDAL: Caught unexpected exception.

(pdal translate readers.gdal Debug) GDAL debug: GDAL: GDALClose(myfile.geojson, this=0x7fc5ca10ebf0)
```

**Expected behavior**
The pipeline should not crash and the *.laz files should be equal.

**System/installation information:**

```
$ pdal --version
------------------------------------------------------------------------------------------
pdal 2.1.0 (git-version: d48e25)
------------------------------------------------------------------------------------------
```

```
uname -a
Darwin Hakons-MacBook-Pro.local 18.7.0 Darwin Kernel Version 18.7.0: Mon Feb 10 21:08:45 PST 2020; root:xnu-4903.278.28~1/RELEASE_X86_64 x86_64
```

Let me know if you need more info!

**Comment by abellgithub** [comment]
I've recreated the issue.

**Comment by abellgithub** [comment]
Resolved by #3097
